pkgs/by-name/ma/material-maker/package.nix

```
@@ -0,0 +1,153 @@
+      hash = "sha256-D4jPQEOKws2JFczh0K55mR08jIBMU6jOFOm6TyFxnt8=";
+    };
+  };
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  inherit pname version;
+
+  src =
+    if stdenvNoCC.isLinux && stdenvNoCC.isx86_64 then
+      sources.linux
+    else if stdenvNoCC.isDarwin then
+      sources.darwin
+    else
+      throw "unsupported platform";
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+  ] ++ lib.optionals stdenvNoCC.isDarwin [ undmg ];
+
+  sourceRoot = ".";
+
+  buildInputs = [
+    libGL
+    libGLU
+    libX11
+    libXcursor
+    libXext
+    libXfixes
+    libXi
+    libXinerama
+    libXrandr
+    libXrender
+  ];
+
+  desktopItems = makeDesktopItem {
+    name = "material-maker";
+    exec = "material-maker";
+    icon = "material-maker";
+    genericName = "Procedural texture generation and 3D model painting tool";
+    desktopName = "Material Maker";
+    comment = finalAttrs.meta.description;
+    terminal = false;
+    categories = [
+      "Graphics"
+      "3DGraphics"
+    ];
+    keywords = [
+      "material"
+      "textures"
+      "paint"
+      "3d"
+    ];
+  };
+
+  installPhase =
+    ''
+      runHook preInstall
+    ''
+    + lib.optionalString stdenvNoCC.isLinux ''
+      install -D -m 755 $src/material_maker.x86_64 $out/opt/material-maker
+      install -D -m 644 $src/material_maker.pck $out/opt/material-maker.pck
+
+      cp -r $src/{library,nodes,environments,meshes,export} $out/opt
+
+      install -D -m 644 $src/doc/_static/icon.png -t $out/share/icons/hicolor/256x256/apps/
+
+      mkdir -p $out/bin
+      ln -s $out/opt/material-maker $out/bin/material-maker
+
+      cp -r $src/doc $out/share
+      cp -r $src/examples $examples
+
+      for output in ${builtins.concatStringsSep " " finalAttrs.meta.outputsToInstall}; do
+        if [ "$output" == "doc" ]; then
+          ln -s $doc/share/doc $out/opt/doc
+        elif [ "$output" == "examples" ]; then
+          ln -s $examples $out/opt/examples
+        fi
+      done
+    ''
+    + lib.optionalString stdenvNoCC.isDarwin ''
+      mkdir -p $out/Applications
+      cp -r ./material_maker.app $out/Applications
+    ''
+    + ''
+      runHook postInstall
+    '';
+
+  outputs = [
+    "out"
+    "doc"
+    "examples"
+  ];
+
+  runtimeDependencies = map lib.getLib [
+    alsa-lib
+    libpulseaudio
+    udev
+  ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    homepage = "https://www.materialmaker.org/";
+    description = "Tool based on Godot Engine that can be used to create textures procedurally and paint 3D models";
+    license = lib.licenses.mit;
+    platforms = lib.platforms.darwin ++ (lib.intersectLists lib.platforms.linux lib.platforms.x86_64);
+    maintainers = with lib.maintainers; [ aurreland ];
+    mainProgram = "material-maker";
+    outputsToInstall = finalAttrs.outputs;
```

It might be better to specify the outputs directly, here:

```suggestion
    outputsToInstall = [
      "out"
      "doc"
      "examples"
    ];
```

If `outputs` change, that shouldn't change the default outputs, right?

---

I was kinda think the same as well, sure.